### PR TITLE
Replaced MPMoviePlayerController with AVPlayerViewController

### PR DESCRIPTION
**Pull request by bennettcheung**
Replaced MPMoviePlayerController with AVPlayerViewController.  Also updated the example video file.

### CODE REVIEW

File: MHVideoPhotoGallery/MMHVideoPhotoGallery/MHGalleryImageViewerViewController/MHGalleryImageViewerViewController.m [hunk 1]
```diff
@@ -1388,10 +1393,15 @@ -(void)moviePlayBackDidFinish:(NSNotification *)notification{
...
     [self stopTimer];
     
-    self.moviePlayer.currentPlaybackTime =0;
+    [self.moviePlayer.player.currentItem seekToTime: kCMTimeZero];
     [self movieTimerChanged:nil];
     [self updateTimerLabels];
     
+}
+
+-(void)newLogError:(NSNotification *)notification{
```
Comment (smasuch) [inline MHVideoPhotoGallery/MMHVideoPhotoGallery/MHGalleryImageViewerViewController/MHGalleryImageViewerViewController.m:1402]: Is this still necessary?

File: MHVideoPhotoGallery/MMHVideoPhotoGallery/MHGalleryImageViewerViewController/MHGalleryImageViewerViewController.m [hunk 2]
```diff
@@ -1485,10 +1495,28 @@ -(MHGalleryViewMode)currentViewMode{
     return MHGalleryViewModeImageViewerNavigationBarShown;
 }
 
+- (void)observeValueForKeyPath:(NSString *)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary *)change
+                       context:(void *)context {
+ 
+    if (context == PlayerStatusContext) {
+           // Do something with the balance…
```
Comment (smasuch) [inline MHVideoPhotoGallery/MMHVideoPhotoGallery/MHGalleryImageViewerViewController/MHGalleryImageViewerViewController.m:1504]: I think this commend can be deleted.

File: MHVideoPhotoGallery/ExampleViewControllerCollectionViewInTableView.m [hunk 3]
```diff
@@ -206,12 +206,10 @@ -(void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPat
...
                     [self setNeedsStatusBarAppearanceUpdate];
                     
-                    MPMoviePlayerController *player = interactiveTransition.moviePlayer;
+                    AVPlayerViewController *controller = interactiveTransition.moviePlayer;
                     
-                    player.controlStyle = MPMovieControlStyleEmbedded;
-                    player.view.frame = cell.bounds;
-                    player.scalingMode = MPMovieScalingModeAspectFill;
-                    [cell.contentView addSubview:player.view];
+                    controller.view.frame = cell.bounds;
```
Comment (smasuch) [inline MHVideoPhotoGallery/ExampleViewControllerCollectionViewInTableView.m:211]: Do we need to deal with any parent/child view controller containment in this case?

File: MHVideoPhotoGallery/MMHVideoPhotoGallery/MHGalleryImageViewerViewController/MHGalleryImageViewerViewController.m (hunk 1 above)
Comment (bennettcheung) [reply MHVideoPhotoGallery/MMHVideoPhotoGallery/MHGalleryImageViewerViewController/MHGalleryImageViewerViewController.m:1402]: No, I was testing something and forgot to remove it.

File: MHVideoPhotoGallery/MMHVideoPhotoGallery/MHGalleryImageViewerViewController/MHGalleryImageViewerViewController.m (hunk 2 above)
Comment (bennettcheung) [reply MHVideoPhotoGallery/MMHVideoPhotoGallery/MHGalleryImageViewerViewController/MHGalleryImageViewerViewController.m:1504]: OK

File: MHVideoPhotoGallery/ExampleViewControllerCollectionViewInTableView.m (hunk 3 above)
Comment (bennettcheung) [reply MHVideoPhotoGallery/ExampleViewControllerCollectionViewInTableView.m:211]: This change is almost the same code as before.  I basically renamed player with controller, perhaps I should have left it unchanged. 
I believe the scaling is done by default.  There is already one set of controls to pause and play video in this example.  That's why I decided to leave it off.

Comment (bennettcheung) [reply MHVideoPhotoGallery/ExampleViewControllerCollectionViewInTableView.m]: I have made another change that should fix this issue.

**Pull request merged**